### ISSUE TITLE
KubernetesClient does not use `sendInitialEvents` for watching

### ISSUE THREAD

**Issue by cwrau** [fabric8io/kubernetes-client#5368]
### Describe the bug

I'm watching a resource like 

```kotlin
kubernetesClient.pods().watch(ListOptionsBuilder().withSendInitialEvents(false).build(), watcher)
```

but I still receive the initial events and looking at the code [here](https://github.com/fabric8io/kubernetes-client/blob/main/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/AbstractWatchManager.java#L304) and [here](https://github.com/fabric8io/kubernetes-client/blob/main/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/BaseOperation.java#L1018-L1054) I don't see any usage of `sendInitialEvents`.

### Fabric8 Kubernetes Client version

6.8.0

### Steps to reproduce

Look at the code or step through a minimal code snippet to see that the option is never used.

### Expected behavior

The option is appended to the URL so I don't get the initial events.

### Runtime

Kubernetes (vanilla)

### Kubernetes API Server version

other (please specify in additional context)

### Environment

Linux

### Additional context

k8s version; 1.27.4

**Comment by shawkins** [comment]
> but I still receive the initial events and looking at the code [here](https://github.com/fabric8io/kubernetes-client/blob/main/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/AbstractWatchManager.java#L304) and [here](https://github.com/fabric8io/kubernetes-client/blob/main/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/BaseOperation.java#L1018-L1054) I don't see any usage of sendInitialEvents.

That is correct the client is currently ignorant of sendInitialEvents.  However this does not match the behavior of what you are describing - this is an alpha feature that requries a feature gate and sending sendIntialEvents=true to get the initial events.  What kubernetes server are you testing against?

Related to #5081

**Comment by cwrau** [comment]
Oh, didn't know that, I just saw the flag and the fact that I was getting an initial list of all resources 😅

I'm just using plain vanilla k8s without any feature gate, installed via kubeadm

This even happens on a 1.25.8 cluster

**Comment by shawkins** [comment]
> Oh, didn't know that, I just saw the flag and the fact that I was getting an initial list of all resources

If you leave the resourceVersion unpopulated or set to 0, then you will be sent the full state as add events https://kubernetes.io/docs/reference/using-api/api-concepts/#semantics-for-watch

Only if you start the watch at a particular resourceVersion will they be omitted.

**Comment by cwrau** [comment]
> > Oh, didn't know that, I just saw the flag and the fact that I was getting an initial list of all resources
> 
> If you leave the resourceVersion unpopulated or set to 0, then you will be sent the full state as add events [kubernetes.io/docs/reference/using-api/api-concepts/#semantics-for-watch](https://kubernetes.io/docs/reference/using-api/api-concepts/#semantics-for-watch)
> 
> Only if you start the watch at a particular resourceVersion will they be omitted.

Ah, I see, then I misunderstood that flag 😅

This can be closed then, thanks for the quick response!

**Comment by shawkins** [comment]
If you don't mind, I'll reopen this to actually add support for sendInitialEvents and at least resourceVersionMatch - I'll need that later in any case for #5081
